### PR TITLE
GTEST: increase timeout in test_ud_slow_timer.ep_destroy

### LINKED ISSUE #3005: failure in ud/test_ud_slow_timer.ep_destroy/0
```
12:25:17 [ RUN      ] ud/test_ud_slow_timer.ep_destroy/0
12:25:19 /scrap/jenkins/workspace/hpc-ucx-pr-3/label/hpc-test-node-upstream/worker/0/contrib/../test/gtest/uct/ib/test_ud_slow_timer.cc:101: Failure
12:25:19 Value of: ucs_ptr_array_lookup(&iface->eps, ep_idx, ud_ep_tmp)
12:25:19   Actual: true
12:25:19 Expected: false
12:25:25 
[  FAILED  ] ud/test_ud_slow_timer.ep_destroy/0, where GetParam() = ud/mlx5_2:1 (7232 ms)
```

### PR DESCRIPTION
**Pull request by evgeny-leksikov**
Fixes #3005

### CODE REVIEW

File: test/gtest/uct/ib/test_ud_slow_timer.cc [hunk 1]
```diff
@@ -97,7 +97,7 @@ UCS_TEST_P(test_ud_slow_timer, ep_destroy, "UD_TIMEOUT=1s") {
     EXPECT_TRUE(ucs_ptr_array_lookup(&iface->eps, ep_idx, ud_ep_tmp));
 
     m_e1->destroy_eps();
-    twait(ucs_time_to_msec(iface->config.peer_timeout * 1.5));
+    twait(ucs_time_to_msec(iface->config.peer_timeout * 3));
```
Comment (yosefe) [inline test/gtest/uct/ib/test_ud_slow_timer.cc]: can we wait for a long time (e.g 60 sec) **or** ep is not present in the array any more?

Comment (evgeny-leksikov) [reply test/gtest/uct/ib/test_ud_slow_timer.cc]: hmm... currently it's unconditional waiting, no reason to wait 60 sec always, will wrap up in a loop with 60 sec timeout.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8184/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8183/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8186/ for details (Mellanox internal link).

File: test/gtest/uct/ib/test_ud_slow_timer.cc [hunk 2]
```diff
@@ -97,7 +97,12 @@ UCS_TEST_P(test_ud_slow_timer, ep_destroy, "UD_TIMEOUT=1s") {
     EXPECT_TRUE(ucs_ptr_array_lookup(&iface->eps, ep_idx, ud_ep_tmp));
 
     m_e1->destroy_eps();
-    twait(ucs_time_to_msec(iface->config.peer_timeout * 1.5));
+    twait(ucs_time_to_msec(iface->config.peer_timeout * 1.1));
+    /* Do more retries if test node is slow */
+    for (size_t i = 0; ucs_ptr_array_lookup(&iface->eps, ep_idx, ud_ep_tmp) &&
+                       (i < 600); ++i) {
+        twait(ucs_time_to_msec(iface->config.peer_timeout * 0.1));
```
[the remark this answers is not in the document]
Comment (shamisp) [reply test/gtest/uct/ib/test_ud_slow_timer.cc]: looks very hacky

File: test/gtest/uct/ib/test_ud_slow_timer.cc [hunk 3]
```diff
@@ -57,6 +57,18 @@ class test_ud_slow_timer : public ud_base_test {
             usleep(1000);
         }
     }
+
+    void wait_for_ep_destroyed(uct_ud_iface_t *iface, uint32_t ep_idx)
+    {
+        ucs_time_t deadline = ucs_get_time() +
+                              ucs_time_from_sec(10) * ucs::test_time_multiplier();
```
Comment (yosefe) [inline test/gtest/uct/ib/test_ud_slow_timer.cc]: i would make it 60 sec.. just in case

Comment (evgeny-leksikov) [reply test/gtest/uct/ib/test_ud_slow_timer.cc]: done

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8207/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8208/ for details (Mellanox internal link).

**Comment by evgeny-leksikov** [comment]
@yosefe ok to merge?

**Pull request merged**